sahi/model.py

```diff
@@ -663,3 +666,161 @@ def _create_object_prediction_list_from_original_predictions(
         object_prediction_list_per_image = [object_prediction_list]
 
         self._object_prediction_list_per_image = object_prediction_list_per_image
+
+
+class TorchVisionDetectionModel(DetectionModel):
+    def load_model(self):
+        try:
+            import torchvision
+        except ImportError:
+            raise ImportError(
+                "torchvision is not installed. Please run 'pip install -U torchvision to use this "
+                "torchvision models'"
+            )
+
+        # set model
+        try:
+            from sahi.utils.torch import torch
+
+            model = self.config_path
+            model.load_state_dict(torch.load(self.model_path))
+            model.eval()
+            model = model.to(self.device)
+            self.model = model
+        except Exception as e:
+            raise Exception(f"Failed to load model from {self.model_path}. {e}")
+
+        # set category_mapping
+        from sahi.utils.torchvision import COCO_CLASSES
+
+        if self.category_mapping is None:
+            category_names = {str(i): COCO_CLASSES[i] for i in range(len(COCO_CLASSES))}
+            self.category_mapping = category_names
+
+    def perform_inference(self, image: np.ndarray, image_size: int = None):
+        """
+        Prediction is performed using self.model and the prediction result is set to self._original_predictions.
+        Args:
+            image: np.ndarray
+                A numpy array that contains the image to be predicted. 3 channel image should be in RGB order.
+            image_size: int
+                Inference input size.
+        """
+        if self.model is None:
+            raise ValueError("model not loaded.")
```

Comment:
@fcakyon Buradaki hata mesajı için önerin var mı daha hoş olabilecek ? Ben bunu placeholder olarak yazmıştım. Eğer yeterli dersen dursun tabi.

Reply:
@kadirnar su sekilde guncelleyebiliriz hata mesajini
```python
raise RuntimeError("Model is not loaded, load it by calling .load_model()")
```